tests/composite_pk/test_filter.py

```diff
@@ -0,0 +1,88 @@
+from django.test import TestCase
+
+from .models import Comment, Tenant, User
+
+
+class CompositePKFilterTests(TestCase):
+    """
+    Test the .filter(), .order_by(), .exclude() methods of composite_pk models.
+    """
+
+    maxDiff = None
+
+    @classmethod
+    def setUpTestData(cls):
+        cls.tenant_1 = Tenant.objects.create()
+        cls.tenant_2 = Tenant.objects.create()
+        cls.user_1 = User.objects.create(tenant=cls.tenant_1, id=1)
+        cls.user_2 = User.objects.create(tenant=cls.tenant_1, id=2)
+        cls.user_3 = User.objects.create(tenant=cls.tenant_2, id=3)
+        cls.comment_1 = Comment.objects.create(id=1, user=cls.user_1)
+        cls.comment_2 = Comment.objects.create(id=2, user=cls.user_1)
+        cls.comment_3 = Comment.objects.create(id=3, user=cls.user_2)
+        cls.comment_4 = Comment.objects.create(id=4, user=cls.user_3)
+        cls.comment_5 = Comment.objects.create(id=5, user=cls.user_1)
+
+    def test_filter_and_count_user_by_pk(self):
+        test_cases = [
+            ({"pk": self.user_1.pk}, 1),
+            ({"pk": self.user_2.pk}, 1),
+            ({"pk": self.user_3.pk}, 1),
+            ({"pk": (self.tenant_1.id, self.user_1.id)}, 1),
+            ({"pk": (self.tenant_1.id, self.user_2.id)}, 1),
+            ({"pk": (self.tenant_2.id, self.user_3.id)}, 1),
+            ({"pk": (self.tenant_1.id, self.user_3.id)}, 0),
+            ({"pk": (self.tenant_2.id, self.user_1.id)}, 0),
+            ({"pk": (self.tenant_2.id, self.user_2.id)}, 0),
+        ]
+
+        for lookup, count in test_cases:
+            with self.subTest(lookup=lookup, count=count):
+                self.assertEqual(User.objects.filter(**lookup).count(), count)
+
+    def test_order_comments_by_pk_asc(self):
+        self.assertEqual(
+            list(Comment.objects.order_by("pk")),
+            [
+                self.comment_1,  # (1, 1)
+                self.comment_2,  # (1, 2)
+                self.comment_3,  # (1, 3)
+                self.comment_5,  # (1, 5)
+                self.comment_4,  # (2, 4)
+            ],
+        )
+
+    def test_order_comments_by_pk_desc(self):
+        self.assertEqual(
+            list(Comment.objects.order_by("-pk")),
+            [
+                # -pk only adds DESC to the last field of the composite key.
```

Comment:
This feels very unintuitive to me. Can we change this so `DESC` is applied to all fields?

Reply:
Yes, I'll look into it, it wasn't so easy to change unfortunately.

Reply:
I found a fix for it !